extensions/HyperVExtension/src/HyperVExtension/Models/VirtualMachineCreation/FileDownloadMonitor.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using HyperVExtension.Extensions;
+using Serilog;
+
+namespace HyperVExtension.Models.VirtualMachineCreation;
+
+/// <summary>
+/// Monitors the download of a file and allows clients to subscribe to the progress of the
+/// download. As new progress comes in the FileDownloadMonitor publishes this progress to
+/// its subscribers.
+/// </summary>
+internal sealed class FileDownloadMonitor
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(FileDownloadMonitor));
+
+    private readonly object _lock = new();
+
+    private readonly List<IProgress<IOperationReport>> _subscriberList = new();
+
+    private readonly Progress<ByteTransferProgress> _progressReporter;
+
+    private bool _downloadInProgress;
+
+    private DownloadOperationReport? _lastSentReport;
+
+    public FileDownloadMonitor(IProgress<IOperationReport> progressProvider)

```

Comment:
Maybe it's just me but "progressProvider" sounds confusing here. Relative to this class it's a subscriber that receives progress events, not provides them. And we add this "progressProvider" to "_subscriberList". I think "progressSubscriber" would be more consistent.

Reply:
sure I can update the name of the parameter